app/serializers/grouped_search_result_serializer.rb

```diff
@@ -4,6 +4,7 @@ class GroupedSearchResultSerializer < ApplicationSerializer
   has_many :posts, serializer: SearchPostSerializer
   has_many :users, serializer: SearchResultUserSerializer
   has_many :categories, serializer: BasicCategorySerializer
+  has_many :topic_categories, serializer: BasicCategorySerializer
```

Comment:
I'm wondering if there is a better name for this attribute to reflect that this is data which we are sideloading and not related to the results. Maybe we can put it under the extras attribute which seems to be the common pattern in our app now. 

However, I do think we should keep the `extras` attribute in the serializer instead of adding it as part of the controller where many spots in our app is doing. 

```
attribute :extras 

def extras
  {
    topic_categories: serialize_data.to_H
  }
end
```
Perhaps something like that?

Reply:
It also looks cleaner to me this way.